src/device/usbd.c

```diff
@@ -1069,17 +1093,20 @@ static bool process_get_descriptor(uint8_t rhport, tusb_control_request_t const
 //--------------------------------------------------------------------+
 // DCD Event Handler
 //--------------------------------------------------------------------+
-TU_ATTR_FAST_FUNC void dcd_event_handler(dcd_event_t const* event, bool in_isr) {
-  bool send = false;
-  switch (event->event_id) {
-    case DCD_EVENT_UNPLUGGED:
-      _usbd_dev.connected = 0;
-      _usbd_dev.addressed = 0;
-      _usbd_dev.cfg_num = 0;
-      _usbd_dev.suspended = 0;
-      send = true;
-      break;
+TU_ATTR_FAST_FUNC void dcd_event_handler(dcd_event_t const * event, bool in_isr)
+{
+  volatile static uint8_t last_event_id = DCD_EVENT_INVALID;
 
+  // Skip the useless repeating bus states
+  if (last_event_id == event->event_id && DCD_EVENT_UNPLUGGED <= event->event_id && DCD_EVENT_RESUME >= event->event_id)
```

Comment:
This will also ignore `DCD_EVENT_SOF`

Reply:
Despite that this can't be merged easily anymore it's also outdated and should contain a bug.

_Maybe next week I can provide the newer version which should work on the current branch._